components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/KeyStoreManagementService.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.security.keystore;
+
+import java.security.cert.X509Certificate;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This service contains the methods to manage certificates of the keystore and client truststore.
+ */
+public interface KeyStoreManagementService {
+
+    /**
+     * Retrieves the list of certificate aliases from the keystore.
+     *
+     * @param filter used to filter the result. Supports sw, ew, eq & co. eg:filter=alias+sw+wso2.
+     * @return the {@link List} of alias.
+     * @throws KeyStoreManagementServerException when retrieving the certificate aliases failed.
+     * @throws KeyStoreManagementClientException when retrieving the certificate aliases failed due to a client error.
+     */
+    List<String> getKeyStoreCertificateAliases(String filter)
```

Comment:
Would it make sense to throw a supertype KeyStoreManagementException instead of server and client errors separately?
Both the KeyStoreManagementServerException and KeyStoreManagementClientException can extend from that class.

Reply:
+1